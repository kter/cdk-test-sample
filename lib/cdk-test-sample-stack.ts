import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { StateMachine, Pass } from 'aws-cdk-lib/aws-stepfunctions';
import { Function, Runtime, Code } from 'aws-cdk-lib/aws-lambda';

export class CdkTestSampleStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id);

    const stateMachine = new StateMachine(this, 'CdkTestSampleStateMachine', {
      definition: new Pass(this, 'StartState'),
    });

    const func = new Function(this, 'LambdaFunction', {
      runtime: Runtime.NODEJS_20_X,
      handler: 'index.handler',
      code: Code.fromAsset('lambda'),
      environment: {
        STATE_MACHINE_ARN: stateMachine.stateMachineArn,
      },
    });
    stateMachine.grantStartExecution(func);

  }
}
