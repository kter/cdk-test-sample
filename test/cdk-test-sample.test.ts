import { App } from 'aws-cdk-lib';
import { Template, Match, Capture } from 'aws-cdk-lib/assertions';
import { CdkTestSampleStack } from '../lib/cdk-test-sample-stack';

describe('CdkTestSampleStack', () => {
  test('Synthesizes', () => {
    const app = new App();
    const stack = new CdkTestSampleStack(app, 'MyTestStack');
    const template = Template.fromStack(stack);

    // Faine-grained assertions Test
    template.hasResourceProperties('AWS::Lambda::Function', {
      Handler: 'index.handler',
      Runtime: 'nodejs20.x',
    });

    // Faine-grained assertions Test
    template.hasResourceProperties('AWS::StepFunctions::StateMachine', {
      DefinitionString: Match.serializedJson(
        {
        StartAt: "StartState",
        States: {
          StartState: {
            Type: "Pass",
            End: true
          }
        }
        }
      )}),

    // Snapshot Test
    // 更新時はテストが失敗するので`npm test -- -u`コマンドを実行しスナップショットをアップデートする
    expect(template.toJSON()).toMatchSnapshot();

    // Fine-grained assertions Test: Capture
    const startAtCapture = new Capture();
    const statesCapture = new Capture();
    template.hasResourceProperties('AWS::StepFunctions::StateMachine', {
      DefinitionString: Match.serializedJson({
        StartAt: startAtCapture,
        States: statesCapture,
      }),
    });
    // 開始状態がStartで始まることを確認
    expect(startAtCapture.asString()).toEqual(expect.stringMatching(/^Start/));
    // statesオブジェクトに開始ステートが存在することを確認
    expect(statesCapture.asObject()).toHaveProperty(startAtCapture.asString());
  });
});


