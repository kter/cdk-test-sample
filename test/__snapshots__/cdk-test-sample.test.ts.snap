// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdkTestSampleStack Synthesizes 1`] = `
{
  "Mappings": {
    "ServiceprincipalMap": {
      "af-south-1": {
        "states": "states.af-south-1.amazonaws.com",
      },
      "ap-east-1": {
        "states": "states.ap-east-1.amazonaws.com",
      },
      "ap-northeast-1": {
        "states": "states.ap-northeast-1.amazonaws.com",
      },
      "ap-northeast-2": {
        "states": "states.ap-northeast-2.amazonaws.com",
      },
      "ap-northeast-3": {
        "states": "states.ap-northeast-3.amazonaws.com",
      },
      "ap-south-1": {
        "states": "states.ap-south-1.amazonaws.com",
      },
      "ap-south-2": {
        "states": "states.ap-south-2.amazonaws.com",
      },
      "ap-southeast-1": {
        "states": "states.ap-southeast-1.amazonaws.com",
      },
      "ap-southeast-2": {
        "states": "states.ap-southeast-2.amazonaws.com",
      },
      "ap-southeast-3": {
        "states": "states.ap-southeast-3.amazonaws.com",
      },
      "ap-southeast-4": {
        "states": "states.ap-southeast-4.amazonaws.com",
      },
      "ca-central-1": {
        "states": "states.ca-central-1.amazonaws.com",
      },
      "cn-north-1": {
        "states": "states.cn-north-1.amazonaws.com",
      },
      "cn-northwest-1": {
        "states": "states.cn-northwest-1.amazonaws.com",
      },
      "eu-central-1": {
        "states": "states.eu-central-1.amazonaws.com",
      },
      "eu-central-2": {
        "states": "states.eu-central-2.amazonaws.com",
      },
      "eu-north-1": {
        "states": "states.eu-north-1.amazonaws.com",
      },
      "eu-south-1": {
        "states": "states.eu-south-1.amazonaws.com",
      },
      "eu-south-2": {
        "states": "states.eu-south-2.amazonaws.com",
      },
      "eu-west-1": {
        "states": "states.eu-west-1.amazonaws.com",
      },
      "eu-west-2": {
        "states": "states.eu-west-2.amazonaws.com",
      },
      "eu-west-3": {
        "states": "states.eu-west-3.amazonaws.com",
      },
      "il-central-1": {
        "states": "states.il-central-1.amazonaws.com",
      },
      "me-central-1": {
        "states": "states.me-central-1.amazonaws.com",
      },
      "me-south-1": {
        "states": "states.me-south-1.amazonaws.com",
      },
      "sa-east-1": {
        "states": "states.sa-east-1.amazonaws.com",
      },
      "us-east-1": {
        "states": "states.us-east-1.amazonaws.com",
      },
      "us-east-2": {
        "states": "states.us-east-2.amazonaws.com",
      },
      "us-gov-east-1": {
        "states": "states.us-gov-east-1.amazonaws.com",
      },
      "us-gov-west-1": {
        "states": "states.us-gov-west-1.amazonaws.com",
      },
      "us-iso-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-iso-west-1": {
        "states": "states.amazonaws.com",
      },
      "us-isob-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-west-1": {
        "states": "states.us-west-1.amazonaws.com",
      },
      "us-west-2": {
        "states": "states.us-west-2.amazonaws.com",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CdkTestSampleStateMachine2573637A": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CdkTestSampleStateMachineRole6756CB99",
      ],
      "Properties": {
        "DefinitionString": "{"StartAt":"StartState","States":{"StartState":{"Type":"Pass","End":true}}}",
        "RoleArn": {
          "Fn::GetAtt": [
            "CdkTestSampleStateMachineRole6756CB99",
            "Arn",
          ],
        },
      },
      "Type": "AWS::StepFunctions::StateMachine",
      "UpdateReplacePolicy": "Delete",
    },
    "CdkTestSampleStateMachineRole6756CB99": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": {
                  "Fn::FindInMap": [
                    "ServiceprincipalMap",
                    {
                      "Ref": "AWS::Region",
                    },
                    "states",
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionBF21E41F": {
      "DependsOn": [
        "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "LambdaFunctionServiceRoleC555A460",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "3f7c7eb464570af916dc9d3206151dac7ede263828a5ded09d3a7f47d34a9248.zip",
        },
        "Environment": {
          "Variables": {
            "STATE_MACHINE_ARN": {
              "Ref": "CdkTestSampleStateMachine2573637A",
            },
            "TEST": "TEST1",
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "LambdaFunctionServiceRoleC555A460",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFunctionServiceRoleC555A460": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionServiceRoleDefaultPolicy32EEEE35": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "states:StartExecution",
              "Effect": "Allow",
              "Resource": {
                "Ref": "CdkTestSampleStateMachine2573637A",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "Roles": [
          {
            "Ref": "LambdaFunctionServiceRoleC555A460",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
